test/packages/bad_configuration_links/manifest.yml

```diff
@@ -79,10 +79,22 @@ policy_templates:
         organization: security
         division: engineering
         team: cloud-security
-        # bad format: missing fields
     configuration_links:
+      # bad format: missing field url
       - title: Security overview
         type: next_step
+        internal: true
+      # bad format: missing field `internal`
+      - title: View Agents
+        url: "/agents"
+        type: next_step
+        content: "Check your agents in Fleet"
+        appId: "fleet"
+      # bad format: `internal: false` should match http or https format
```

Comment:
I'm not sure how to enforce this rule. I found the `pattern` field but can I enforce it conditionally (based on `internal` field)?

Reply:
I think you would need to use `anyOf` or `oneOf`. Something like this:
```
          internal:
            description: Set to true for internal kibana links, false for external links. Defaults to false.
            type: boolean
            oneOf:
              - const: true
                pattern: "^/"
              - const: false
                pattern: "^https?://"
```

Reply:
I think it can be used `if - then` definitions here to set those conditions

I've tried to set these conditions in the manifest definition:
```diff
 $ git diff
diff --git spec/integration/manifest.spec.yml spec/integration/manifest.spec.yml
index cf39a4f..bd79026 100644
--- spec/integration/manifest.spec.yml
+++ spec/integration/manifest.spec.yml
@@ -235,6 +235,30 @@ spec:
             description: Set to true for internal kibana links, false for external links. Defaults to false.
             type: boolean
             default: false
+        allOf:
+          - if:
+              properties:
+                internal:
+                  const: true
+            then:
+              required:
+                - appId
+              properties:
+                url:
+                  type: string
+                  pattern: '^/'
+          - if:
+              properties:
+                internal:
+                  const: false
+            then:
+              not:
+                required:
+                  - appId
+              properties:
+                url:
+                  type: string
+                  pattern: '^http(s)?://'
         required:
         - title
         - url
```

Adding those definitions there are some errors, that probably are legit. 

These conditions enforce:
- a pattern for `url` depending on the `internal` value
- `appId` is set required or not depending on the `internal` value.
Is there any other condition that should be added ?

Reply:
Thanks, I'm switching to use format `kbn:...` for internal links as per below [discussion](https://github.com/elastic/package-spec/pull/844#discussion_r1881798689). Example `kbn:app/integrations/browse`